modules/platform/nuxeo-apidoc-server/nuxeo-apidoc-core/src/main/java/org/nuxeo/apidoc/documentation/DocumentationHelper.java

```diff
@@ -134,15 +148,54 @@ public static String getHtml(String doc) {
      * Makes sure no passwords are embedded in the XML.
      */
     public static String secureXML(String xml) {
-        if (xml == null || !xml.contains("assword")) {
+        if (StringUtils.isBlank(xml)) {
             return xml;
         }
-        xml = xml.replaceAll("<([a-zA-Z]*[pP])assword>[^<]*</([a-zA-Z]*)assword>", "<$1assword>********</$2assword>");
-        // attributes: nuxeo-core-auth
-        xml = xml.replaceAll("([a-zA-Z]*[pP])assword=\"[^\"]*\"", "$1assword=\"********\"");
-        // property: default-repository-config
-        xml = xml.replaceAll("([a-zA-Z]*[pP])assword\">[^<]*<", "$1assword\">********<");
-        return xml;
+        String res = xml;
+        List<String> keywords = getKeywordList(SECURE_KEYWORDS_PROPERTY, DEFAULT_SECURE_KEYWORDS);
+        List<String> whitelist = getKeywordList(WHITELISTED_KEYWORDS_PROPERTY, DEFAULT_WHITELISTED_KEYWORDS);
+        for (String kw : keywords) {
+            if (res.contains(kw)) {
+                for (String pattern : List.of(
+                        // node startswith
+                        String.format("(?<start><(?<key>\\w*%s)\\s*>)[^<]*(?<end></\\w*%s>)", kw, kw),
```

Comment:
I would have done the rewriting using StAX instead of ad hoc regexps...

Reply:
As discussed, i hesitated, and then changed my mind. I will see that again with https://jira.nuxeo.com/browse/NXP-28948 if you're ok with it as i will need to handle cases where XML would not be valid (while regexps would still work)

Reply:
After more discussion, will use StAX as primary parser, to handle more use cases, and fallback on regexps in case of invalid XML